### ISSUE TITLE
Cannot change strategy of configuration ':apt' after it has been resolved

### ISSUE THREAD

**Issue by yissachar** [tbroyer/gradle-apt-plugin#31]
Cross-posting [this issue](https://github.com/akhikhl/gretty/issues/284) here, since I'm not sure if it's a Gretty issue or an issue with your APT plugin:

When using the Gretty plugin in combination with the APT plugin, I receive the following error:

>Cannot change strategy of configuration ':apt' after it has been resolved

in Gradle < 3.0 this was just a warning and I was able to proceed anyway, but in Gradle 3.0 this is an error that prevents me from building the project.

I am using Gretty 1.4.0 and Gradle-APT-Plugin 0.8 (both latest versions). Simple build.gradle that you can use to reproduce this:

```
apply plugin: 'java'
apply plugin: 'net.ltgt.apt'
apply plugin: 'org.akhikhl.gretty'

buildscript {
    repositories {
        maven { url "https://plugins.gradle.org/m2/" }
    }
    dependencies {
        classpath "net.ltgt.gradle:gradle-apt-plugin:0.8"
        classpath 'org.akhikhl.gretty:gretty:1.4.0'
    }
}
```

**Comment by tbroyer** [comment]
Note: if you apply the `net.ltgt.apt` plugin **after** the `org.akhikhl.gretty` plugin, it'll work.

This is due to the order the `afterEvaluate` _hooks_ are called, the one from the `net.ltgt.apt` plugin triggering a resolve of the `apt` configurations. I'll see if I can change the plugin to defer this, maybe, to a `first` action on the compile tasks.

In the meantime, at least, you have now have a workaround.

**Comment by yissachar** [comment]
That works, thanks!

**Comment by tbroyer** [comment]
Should be fixed by 6c137f71f069b621ed7834314e59327bde289495

**Comment by tbroyer** [comment]
Just released v0.9 with the fix btw
